pkg/controller/mutators/stats_reporter_test.go

```diff
@@ -37,31 +37,31 @@ func TestReportMutatorIngestionRequest(t *testing.T) {
 	}
 
 	// Count test
-	row, err := checkData(mutatorIngestionCountMetricName, wantRowLength)
+	rows, err := checkData(mutatorIngestionCountMetricName, wantRowLength)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	gotCount, ok := row.Data.(*view.CountData)
+	gotCount, ok := rows[0].Data.(*view.CountData)
```

Comment:
why do we only care about index 0?

Reply:
This is some confusing code that I've extended to return multiple rows instead of just returning the first one.  You'll see in the [old version of checkData](https://github.com/open-policy-agent/gatekeeper/pull/1714/files#diff-fc62bee533cdd4a60bd41c0a636c0afe9a0536c0f0848cb0cf85d7983f76527aL88) that it was just doing that _in_ the function, and now does it _outside_ of it.

These tests are due for an overhaul, and that would be worthy cleanup work.  If that's worth doing, I think it'd be best done in a separate PR that builds functionality for validating rows of metric data more effectively and using that logic across _all_ the metric unit tests.

SGTY @ritazh ?